fairlearn/metrics/_annotated_metric_function.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+from typing import Callable, Dict, List
+
+import pandas as pd
+
+
+class AnnotatedMetricFunction:
+    """Wrapper for functions, to give them args and kwargs."""
+
+    def __init__(self,
+                 *,
+                 func: Callable,
+                 args: List[str] = None,
+                 kwargs: Dict[str, str] = None):
+        self._func = func
+        self._args = ['y_true', 'y_pred']
+        if args is not None:
+            self._args = args
+        self._kwargs = dict()
+        if kwargs is not None:
+            self._kwargs = kwargs
+
+    @property
+    def func(self) -> Callable:
+        """Return the wrapped function."""
+        return self._func
+
```

Comment:
could we not have properties for these and just store them in public attributes instead of private ones? It makes the code much smaller and simpler.

Reply:
_sobs_
Removed.